executor-service/src/main/java/executor/service/service/scenario/ScenarioHttpService.java

```diff
@@ -27,6 +29,28 @@ public void saveScenarios(List<Scenario> scenarios) {
         scenarioQueue.addAll(scenarios);
     }
 
+    @Override
+    public ResponseEntity<Scenario> getScenarioByUsername(
+            final String username, final String scenarioName) {
+        Scenario scenario = getScenariosByUsername(username)
+                    .stream()
+                    .filter(s -> s.getUsername().equals(username))
+                    .findFirst()
+                    .orElse(null);
+        return scenario != null
+               ? new ResponseEntity<>(scenario, HttpStatus.OK)
+               : new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
There may be multiple user's Scenarios with the same name, so we could return 
`List<Scenario>getScenarioByUsername(final String username, final String scenarioName)`:
```suggestion
    public List<Scenario> getScenarioByUsername(
            final String username, final String scenarioName) {
        return getScenariosByUsername(username)
                    .stream()
                    .filter(s -> s.getName().equals(scenarioName))
                    .collect(Collectors.toList());
    }
```